10.11/docker-entrypoint.sh

```diff
@@ -227,8 +227,18 @@ _mariadb_version() {
 docker_init_database_dir() {
 	mysql_note "Initializing database files"
 	installArgs=( --datadir="$DATADIR" --rpm --auth-root-authentication-method=normal )
-	# "Other options are passed to mariadbd." (so we pass all "mariadbd" arguments directly here)
-	mariadb-install-db "${installArgs[@]}" "${@:2}" \
+
+	mariadbdArgs=()
+	for arg in "${@:2}"; do
+		# Check if the argument contains whitespace
+		if [[ ! "$arg" =~ [[:space:]] ]]; then
+			mariadbdArgs+=("$arg")
+		else
+			mysql_warn Not passing argument \'$arg\' to mariadb-install-db because mariadb-install-db does not support arguments with whitespace.
```

Comment:
Should this be warning or error? Warnings could be silently ignored in the automation and lead to potential failures.

Reply:
Not sure if you are asking me ...
... but I would agree with you.

Reply:
I consider an error as something that cannot proceed and a warning is something that might be impacting.  As the original example showed there are some things that make little difference to the way mariadb-install-db processes.

Regardless I'm hoping that this is only a temporary change until the server implementation is made cleaner with some actual shell programming expertise.

Reply:
LGTM but I would rewrite the if:

```bash
if [[ "$arg" =~ [[:space:]] ]]; then
  warning
else
  OK
fi
```
seems much more simple to understand.